packages/core/src/browser/frontend-application-module.ts

```diff
@@ -64,4 +65,7 @@ export const frontendApplicationModule = new ContainerModule(bind => {
         return messageService;
     }).inSingletonScope();
     bind(MessageService).toSelf().inSingletonScope();
+
+    bind(CommandContribution).to(CoreContribution).inSingletonScope();
+    bind(KeybindingContribution).to(CoreContribution).inSingletonScope();
```

Comment:
in such config you create 2 separate instances of `CoreContribution`, better:
```ts
bind(CoreContribution).toSelf().inSingletonScope();
[CommandContribution, KeybindingContribution].forEach(serviceIdentifier =>
  bind(serviceIdentifier).toDynamicValue(ctx => ctx.container.get(CoreContribution)).inSingletonScope()
);
```

Reply:
I'm confused here how will it be 2 instances of CoreContribution if it's a Singleton ?

Reply:
one when `CommandContribution` is requested, another then `KeybindingContribution`

Reply:
OK my bad I read it as 2x CoreContribution which didn't make sense :)